server/memphis_handlers_user_mgmt.go

```diff
@@ -136,13 +136,47 @@ func updateDeletedUserResources(user models.User) error {
 	return nil
 }
 
-func validateUsername(username string) error {
-	re := regexp.MustCompile("^[a-z0-9_.-]*$")
+func removeDeletedUsersResources(tenantName string) error {
+	err := db.RemoveProducersOfDeletedUsers(tenantName)
+	if err != nil {
+		return err
+	}
+
+	err = db.RemoveConsumersOfDeletedUsers(tenantName)
+	if err != nil {
+		return err
+	}
+
+	err = db.RemoveConnectionsOfDeletedUsers(tenantName)
+	if err != nil {
+		return err
+	}
+
+	err = db.RemoveSchemaVersionsOfDeletedUsers(tenantName)
+	if err != nil {
+		return err
+	}
+
+	err = db.RemoveSchemasOfDeletedUsers(tenantName)
+	if err != nil {
+		return err
+	}
+
+	err = db.RemoveStationsOfDeletedUsers(tenantName)
```

Comment:
I think that stations should be deleted at the end because it is a foreign key of audit logs for example

Reply:
You should also need to delete NATS streams and NATS consumers, take a reference from delete station/delete consumer